modules/c++/sys/source/OSUnix.cpp

```diff
@@ -44,6 +44,26 @@
 #include "sys/OSUnix.h"
 #include "sys/File.h"
 
+
+namespace
+{
+void linkStat(const std::string& pathname, struct stat& linkStatus)
+{
+    if (lstat(pathname.c_str(), &linkStatus) != 0)
+    {
+        throw except::Exception(Ctxt(strerror(errno)));
+    }
+}
+
+void readlinkOrError(const std::string& pathname, char* buffer)
```

Comment:
Should return the result as a `std::string` here rather than taking in a `char*`.  That way the caller doesn't need to know any details about PATH_MAX or mess with any buffers.